modules/frontend/querysharding.go

```diff
@@ -53,7 +56,7 @@ func (s shardQuery) Do(r *http.Request) (*http.Response, error) {
 
 	// only need to initialise boundaries once
 	if len(s.blockBoundaries) == 0 {
-		s.blockBoundaries = createBlockBoundaries(s.queryShards)
+		s.blockBoundaries = createBlockBoundaries(s.queryShards - 1) // one shard will be used to query ingesters
```

Comment:
if this is initialized once let's move it to the `ShardingWare` function above where the struct is created instead of relying on this nil check.